app/variablesmanager.cpp

```diff
@@ -0,0 +1,41 @@
+#include "variablesmanager.h"
+
+#include "qgsexpressioncontextutils.h"
+
+VariablesManager::VariablesManager( MerginApi *merginApi, QObject *parent )
+  : QObject( parent )
+  , mMerginApi( merginApi )
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_url" ),  mMerginApi->apiRoot() );
+
+  QObject::connect( mMerginApi, &MerginApi::apiRootChanged, this, &VariablesManager::apiRootChanged );
+  QObject::connect( merginApi, &MerginApi::authChanged, this, &VariablesManager::authChanged );
+}
+
+void VariablesManager::apiRootChanged()
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_url" ),  mMerginApi->apiRoot() );
+}
+
+void VariablesManager::authChanged()
+{
+  QgsExpressionContextUtils::setGlobalVariable( QStringLiteral( "mergin_username" ),  mMerginApi->merginUserName() );
```

Comment:
What if user logs out - or fails to log in - will be the username set?

Reply:
if a user logs out, authChanged is called anyway. If a user wants to log in, but fails, a user had to be log out before and therefore username is empty and will be empty after it fails.

Question - Use case:  logged in user who suddenly lost connection. Then a user wants to reload list of project and but fails to authorised (authFailed signal) - do we want to keep username or not?